src/services/posts.js

```diff
@@ -88,12 +89,30 @@ const queryPostTags = async({tag}) => {
 	)
 	return feed.rows
 }
+const assignTagToPost = async({postId , tagId}) => {
+	const result = await db.query(
+		// eslint-disable-next-line max-len
+		`INSERT INTO posts_tags ("postId","tagId") 
+		SELECT $1,$2 WHERE NOT EXISTS 
+		(SELECT * FROM posts_tags WHERE "postId" = $1 AND "tagId" = $2) 
+		RETURNING *`,
+		[postId, tagId]
+	)
+	const postsTags = result.rows[0]
+	if (!postsTags) {
+		const errorMessage = TagAlreadyAssignedToPostError.message
```

Comment:
@beblicarl this way of assigning errors a bit verbose and I think we're doing it in more than 5 places now so let's refactor this into a function

1. create a `custom-error.js` file in the `src/lib` folder
2. create a function called `customError` which takes `name` and `message` as parameters and now does this logic there and export it. something like

```js
const customError = ({name, message}) => {
  const error = new Error(message)
  err.name =  name
  return err
}
```
4.  import the `customError` here and use that instead

so your service code will now look like
```js
throw customError(TagAlreadyAssignedToPostError)
```